### ISSUE TITLE
unstacking and (built-in) dtype

### ISSUE THREAD

**Issue by lfaucheux** [hgrecco/pint-pandas#70]
It looks like there is some kind of dtype inference when unstacking data isn't it ? This may be a problem with my environment (corporate one, I cannot control, not really sane), and I am not sure that you will be able to reproduce the following issue. Say you have 

```
>>> pd.DataFrame({"ang_vel": pd.Series([1., 2.1, 2., 3.], dtype="pint[rpm]")}).unstack()
ang_vel  0    1.0
         1    2.1
         2    2.0
         3    3.0
dtype: pint[revolutions_per_minute]
```

Everything is fine. Now consider the following:

```
>>> pd.DataFrame({"ang_vel": pd.Series([1, 2.1, 2., 3.], dtype="pint[rpm]")}).unstack()
>>> #                                   ^
ang_vel  0    1
         1    2
         2    2
         3    3
dtype: pint[revolutions_per_minute]
```

This may even be a desired behaviour 0o. Which I doubt :) 

Thx. Pint is amazing anyway.

**Comment by andrewgsavage** [comment]
What version are you on? This should be fixed in 0.2, following https://github.com/hgrecco/pint-pandas/pull/62

I get this:
```
pd.DataFrame({"ang_vel": pd.Series([1, 2.1, 2., 3.], dtype="pint[rpm]")}).unstack()
​ang_vel  0    1.0
         1    2.1
         2    2.0
         3    3.0
dtype: pint[revolutions_per_minute]
```

**Comment by lfaucheux** [comment]
Indeed `0.16.1` (with `pint-pandas-fork==0.1.dev0` incidentally).
